packages/ragbits-cli/src/ragbits/cli/app.py

```diff
@@ -0,0 +1,76 @@
+import json
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any
+
+import typer
+from pydantic import BaseModel
+from rich.console import Console
+from rich.table import Table
+
+
+class OutputType(Enum):
+    """Indicates a type of CLI output formatting"""
+
+    TEXT = "text"
+    JSON = "json"
+
+
+@dataclass()
+class CliState:
+    """A dataclass describing CLI state"""
+
+    output_type: OutputType = OutputType.TEXT
+
+
+class CLI(typer.Typer):
+    """A CLI class with output formatting"""
+
+    def __init__(self, *args: Any, **kwargs: Any):  # noqa: ANN401
+        super().__init__(*args, **kwargs)
+        self.state: CliState = CliState()
+        self.console: Console = Console()
+
+    def set_output_type(self, output_type: str) -> None:
+        """
+        Set the output type in the app state
+        Args:
+            output_type: str
+        Raises:
+            ValueError - if the output_type is not `json` or `text`
+        """
+        if output_type not in ["text", "json"]:
+            raise ValueError("Output type must be either 'text' or 'json'")
+        if output_type == "json":
+            self.state.output_type = OutputType.JSON
+        if output_type == "text":
+            self.state.output_type = OutputType.TEXT
+
+    def print_output(self, data: list[BaseModel] | BaseModel) -> None:
+        """
+        Process and display output based on the current state's output type.
+
+        Args:
+            data: list of ditionaries or list of pydantic models representing output of CLI function
+        """
+        if isinstance(data, BaseModel):
+            data = [data]
+        if len(data) == 0:
+            raise ValueError("At last one element needs to be passed")
```

Comment:
In my mind an empty list is a normal use case that should be supported without raising an exception (that's what I meant by my previous comment - the previous version of the code would result in `KeyError`).

For example `ragbits vector-store list` and `ragbits vector-store query` may result in 0 results - that's entirely normal. I believe that in json mode this should print an empty list and in text mode - some non-error message (Like "No results")

Reply:
done